session.go

```diff
@@ -0,0 +1,104 @@
+package sdk
+
+import (
+	"context"
+	"errors"
+
+	"github.com/cosmos/gogoproto/grpc"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	grpcoptions "google.golang.org/grpc"
+)
+
+// SessionClient is the interface to interact with the on-chain session module.
+//
+// For example, it can be used to get the current session for a given application
+// and service id at a given height.
+type SessionClient struct {
+	PoktNodeSessionFetcher
+}
+
+// GetSession returns the session with the given application address, service id and height.
+func (s *SessionClient) GetSession(
+	ctx context.Context,
+	appAddress string,
+	serviceId string,
+	height int64,
+) (session *sessiontypes.Session, err error) {
+	if s.PoktNodeSessionFetcher == nil {
+		return nil, errors.New("PoktNodeSessionFetcher not set")
+	}
+
+	req := &sessiontypes.QueryGetSessionRequest{
+		ApplicationAddress: appAddress,
+		Service:            &sharedtypes.Service{Id: serviceId},
+		BlockHeight:        height,
+	}
+
+	// TODO_DISCUSS: Would it be feasible to add a GetCurrentSession, supported by the underlying protocol?
+	// It seems likely that GetSession will almost always be used to get the session matching the latest height.
+	// If this is a correct assumption, the aforementioned support from the protocol could simplify fetching the current session.
+	// In addition, the current session that is being returned could include the latest block height, reducing the number of
+	// SDK calls needed for sending relays.
+	res, err := s.PoktNodeSessionFetcher.GetSession(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	return res.Session, nil
+}
+
+// TODO_DISCUSS: ServiceEndpoints can be replaced by a method on the Session struct.
+// TODO_DISCUSS: Consider using a custom type, defined as a string, as supplier address.
+// This can help enforce type safety by requiring explict type casting of a string before it can be used as a SupplierAddress.
+//
+// ServiceEndpoints returns the supplier endpoints assigned to a session for the given service id.
+//
+// The returned value is a map of supplierId to the list of endpoints.
+func ServiceEndpoints(
+	session *sessiontypes.Session,
+	serviceId string,
+) map[string][]*sharedtypes.SupplierEndpoint {
+	supplierEndpoints := make(map[string][]*sharedtypes.SupplierEndpoint)
```

Comment:
I'd rather keep the `SessionSuppliers` struct and use it to return the available suppliers data to be used by `SendRelay`

* It will ensure that a matching `SupplierAddress/SupplierUrl` is provided to `SendRelay` instead of potentially dissociated ones.
* It also returns all the supplier urls while retaining the corresponding `RPCType` info.
* A matching `SessionHeader` is always accompanying the `Supplier` provided.

Reply:
Great point, thank you.
Since endpoint selection is a feature that is of interest to Gateways, and is always done in the context of a session, a better approach, compared to the first version of the PR, is to use a wrapper struct. I have updated the PR with a first version of this approach. PTAL.